pkg/apis/istio/v1alpha1/istioinstaller_types.proto

```diff
@@ -0,0 +1,441 @@
+    // by the IstioInstaller e.g. prometheus, grafana, kiali.
+    repeated OperatorSpec external_operators = 50;
+
+    // Global k8s defaults. Can be overridden per component.
+    KubernetesResourcesSpec k8s_defaults = 80;
+
+    // URL for the install package. e.g.
+    //   http://github.com/istio/istio/releases/install/1.1.2
+    //   http://github.com/istio/istio/releases/install/lts
+    //   file://tmp/istio-installer/nightly
+    // Uses compiled in install package by default.
+    string custom_package_path = 100;
+    // Root for docker image paths e.g. docker.io/istio-release.
+    // Releases are published to docker hub under 'istio' project.
+    // Daily builds from prow are on gcr.io, and nightly builds from circle on docker.io/istionightly
+    string hub = 101;
+    // Version tag for docker images e.g. 1.0.6
+    string tag = 102;
+}
+
+// Configuration options for traffic management.
+message TrafficManagementFeatureSpec {
+    // Selects whether traffic management is installed.
+    // Must be enabled to enable any sub-component.
+    google.protobuf.BoolValue enabled = 1;
+    // Namespace that all traffic management components are installed into.
+    string namespace = 2;
+
+    // Component specific config.
+    PilotComponentSpec pilot = 30;
+    ProxyComponentSpec proxy = 31;
+    SidecarInjectorComponentSpec sidecar_injector = 32;
+}
+
+// Configuration options for the policy and telemetry feature.
+message PolicyTelemetryFeatureSpec {
+    // Default behavior of the sidecar for handling outbound traffic from the application.
+    enum OutboundPolicy {
+        // Outbound traffic to unknown destinations will be allowed, in case there are no
+        // services or ServiceEntries for the destination port
+        ALLOW_ANY = 0;
+        // Restrict outbound traffic to services defined in the service registry as well
+        // as those defined through ServiceEntries
+        REGISTRY_ONLY = 1;
+    }
+
+    // If set, allows traffic in cases when the mixer policy service cannot be reached.
+    google.protobuf.BoolValue policy_check_fail_open = 10;
+    // Selects the outbound traffic policy mode.
+    OutboundPolicy outbound_traffic_policy_mode = 11;
+
+    PolicyComponentSpec policy = 50;
+    TelemetryComponentSpec telemetry = 51;
+}
+
+// Configuration options for security feature.
+message SecurityFeatureSpec {
+    // Selects whether security feature is installed. Must be set for any sub-component to be installed.
+    google.protobuf.BoolValue enabled = 1;
+
+    // Namespace that security components are installed into.
+    string namespace = 2;
+
+    // Enables MTLS for control plane.
+    // Will result in delays starting the pods while secrets are propagated, not recommended for tests.
+    google.protobuf.BoolValue control_plane_mtls = 13;
+
+    // Enables MTLS for service to service traffic.
+    google.protobuf.BoolValue data_plane_mtls = 14;
+
```

Comment:
what about mTLS strict?  i feel that is the most important config for security.

Reply:
That's exactly what this is. The documentation is misleading here and in the values.yaml description, I'll update it.

Reply:
Thx for the update, makes sense now.  could we use consistent naming? control_plane_mtls -> control_plane_mtls_strict or change data_plane_mtls_strict to data_plane_mtls?   IMHO, strict is our impl detail, people expect mtls to be mtls enforced.

Reply:
i think control_plane_mtls_strict would imply this can be permissive, which is not true. it's either off, or strict. 
otoh, for the data plane, this knob controls whether the setting is strict or permissive. 
as such i think the field names are correct in their current form - wdyt?

Reply:
are you saying control_plane_mtls can be either permissive or enforced mTLS? i am still concerned with the inconsistent naming.

Reply:
you are right, that is quite confusing but i see that is what is implemented today now.

Reply:
so i think we agree? meaning that the functionality is different and therefore it makes sense to have different names for the fields here. feel free to reopen this thread if we don't, in fact, agree :-)